test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/ServerTests.cs

```diff
@@ -88,6 +88,30 @@ public class ServerTests
         }
 
         [ConditionalFact]
+        public async Task Server_ShutdownDurringLongRunningRequest_TimesOut()
+        {
+            Task<string> responseTask;
+            var received = new ManualResetEvent(false);
+            bool? shutdown = null;
+            var waitForShutdown = new ManualResetEvent(false);
+            string address;
+            using (Utilities.CreateHttpServer(out address, httpContext =>
+            {
+                received.Set();
+                shutdown = waitForShutdown.WaitOne(TimeSpan.FromSeconds(15));
+                httpContext.Response.ContentLength = 11;
+                return httpContext.Response.WriteAsync("Hello World");
+            }))
+            {
+                responseTask = SendRequestAsync(address);
+                Assert.True(received.WaitOne(TimeSpan.FromSeconds(10)));
+            }
+            Assert.False(shutdown.HasValue);
+            waitForShutdown.Set();
+            await Assert.ThrowsAsync<HttpRequestException>(async () => await responseTask);
```

Comment:
Are there any specific things you can check in the exception to make sure it failed for the right reason?